sdk/storage/Azure.Storage.Blobs/src/Models/BlobUploadDirectoryOptions.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Azure.Storage.Blobs.Models
+{
+    /// <summary>
+    /// Blob Upload Directory Options
+    /// <see cref="BlobUploadOptions"/> for more options.
+    /// </summary>
+    public class BlobUploadDirectoryOptions : BlobUploadOptions
+    {
+        /// <summary>
+        /// Optional Blob Type to Upload. Defaults to upload Block Blobs.
+        /// </summary>
+        public BlobType blobType { get; set; }
```

Comment:
I'd focus on block blobs first. We recently learnt that providing resilient upload for page blobs for instance is quite challenging. Do we know how popular is to upload non-block blobs in track 1 DM lib?

Reply:
I'm not sure how popular it is, if it is a feature in track1 then I'm sure some customers may want the ability to set it.